src/components/ZipCodeInput.js

```diff
@@ -0,0 +1,32 @@
+import React, { useEffect, useState } from "react";
+
+export default function ZipCodeInput({name, placeholder, value, onChanged}) {
+    const [inputValue, setInputValue] = useState("");
+
+    const inputChanged = (e) => {
+        e.preventDefault();
+        if (e.target.validity.valid || e.target.value === "") {
+            setInputValue(e.target.value);
+            onChanged(e.target.value);
+        }
+    };
+
+    useEffect(() => {
+        setInputValue(value);
+        onChanged(value);
+    }, [value, onChanged]);
+
+    return(
+        <div>
+            {name}<br/>
+            <input
+                name={name.split(" ").join("_").toLowerCase()}
+                type="tel"
+                pattern="[0-9-]*"
```

Comment:
why not something like?

```suggestion
                pattern="^(\d{5}(?:\-\d{4})?)$"
```

https://regexlib.com/Search.aspx?k=us+zip+code&c=-1&m=-1&ps=20&AspxAutoDetectCookieSupport=1